### ISSUE TITLE
UltiSnips not inserting snippets---keymapping conflicts: autopair, deoplete, UltiSnips.

### ISSUE THREAD

**Issue by yzhuang** [liuchengxu/space-vim#138]
#### Environment

- OS: OS X 10.12.4
- vim: N/A
- neovim: 0.1.7

#### Problem Description

Added the `code-snippets` layer and the \<CR\> key does not insert snippets. 
Examining imap shows that both autopair and deoplete changed the behavior of the \<CR\> key.

**Comment by liuchengxu** [comment]
`code-snippets` Layer doesn't work? Have you tested in insert mode? It won't be loaded until you first time enter insert mode.

**Comment by yzhuang** [comment]
It works in the sense that I can see the snippets.
 
But when the snippet is selected, typing <CR> does not insert the snippet, but inserts a newline instead.

**Comment by liuchengxu** [comment]
Thanks. I'll look into this tonight.

**Comment by yzhuang** [comment]
To help repro, I am using the following layers:

```
    Layer 'fzf'
    Layer 'unite'
    Layer 'better-defaults'
    Layer 'file-manager'

    Layer 'tmux'
    Layer 'programming'
    Layer 'code-snippets'
    Layer 'auto-completion'
    Layer 'python'
```

**Comment by liuchengxu** [comment]
The ultisnips and deoplete.nvim are indeed problematic. I  need more time to delve into this.

Since you only use neovim,  I think https://github.com/roxma/nvim-completion-manager is also worthy of giving a try. It supports ultinips and neosnippets.

**Comment by yzhuang** [comment]
Figured out the solution to this problem, but the solution is not very ideal: the ordering of the layers matter.

The \<CR\> key does not work correctly in insert mode with this:
```
    Layer 'code-snippets'
    Layer 'auto-completion'
```

But works correctly with this:
```
    Layer 'auto-completion'
    Layer 'code-snippets'
```

For now, I just swapped the order of these two layers in my .spacevim file.
I am closing this issue for now.  Up to you to decide whether you want to fix the conflict.
